core/src/main/java/org/acegisecurity/context/SecurityContext.java

```diff
@@ -27,46 +27,67 @@
 import edu.umd.cs.findbugs.annotations.NonNull;
 import hudson.model.User;
 import hudson.security.ACL;
+import java.io.Serializable;
 import org.acegisecurity.Authentication;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
 
 /**
  * @deprecated Use {@link ACL#as(User)} or {@link org.springframework.security.core.context.SecurityContext}
  */
 @Deprecated
-public interface SecurityContext {
+public interface SecurityContext extends Serializable {
```

Comment:
Note that just this line suffices to make the tests pass. I refactored the inner classes to member classes since it is not a good practice to use inner classes for anything that might be serialized.